test/bio_write_test.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <stdio.h>
+#include <string.h>
+#include <openssl/evp.h>
+#include <openssl/bio.h>
+#include <openssl/rand.h>
+#include <openssl/bio.h>
+
```

Comment:
This file should be using the test framework more fully, i.e. using the TEST_ macros as defined in testutil.h. Take a look at the other tests for examples.

Reply:
This comment is still relevant.